invisible_cities/cities/detsim.py

```diff
@@ -123,6 +125,8 @@ def detsim(*, files_in, file_out, event_range, print_mod, compression,
            detector_db, run_number, s1_lighttable, s2_lighttable, sipm_psf,
            buffer_params, physics_params, rate):
 
+    buffer_params["max_time"] = copy.deepcopy(check_max_time(buffer_params["max_time"], 
```

Comment:
This is wrong. Let me show an example of what goes on with mutable objects like this:
```
def func(a_dict):
    a_dict["b"] = "world"

d = {"a": "hello"}
func(d)
print(d)
```

As you can see, after calling func you get `d` modified. This is because dictionarys are mutable objects: inside the funcion you are not really having a new `a_dict` variable, but the passed `d`.

Then you may notice that your change does nothing different from what it was implemented before, `buffer_params` is still the same dictionary and is being modified inside the function, which we want to avoid. 
If you look at the next line (130) you can see how this is avoided for `physics_params`, which is also a dictionary. I wanted you to apply this same approach, ie creating a new dictionary inside the function itself, being a copy of the original one.

Reply:
Okay, we can do the same as in the lines below
```python
buffer_params_ = buffer_params.copy()
buffer_params["max_time"] = check_max_time(buffer_params_["max_time"], buffer_params_["length"])
```
and could this be better too?
```python
def check_max_time(max_time, buffer_length):
    """ Maybe this is better to avoid 
        substitution in all cases """
    if max_time < buffer_length:
        ## WARNING ##
        return buffer_length
    else:
        return max_time
```

Reply:
Yes, it is better.

Caution here:
```
buffer_params_ = buffer_params.copy()
buffer_params["max_time"] = check_max_time(buffer_params_["max_time"], buffer_params_["length"])
```
must be
```
buffer_params_ = buffer_params.copy()
buffer_params_["max_time"] = check_max_time(buffer_params_["max_time"], buffer_params_["length"])
```

Reply:
Ah, right. But if we do
```python
buffer_params_["max_time"] = check_max_time(...)
```
then we need to update the main dictionary `buffer_params`:
```python
buffer_params = buffer_params_
```
Something like this?

Reply:
Why?

Reply:
Because I was still confused. @mmkekic helped me realize what is done with the other mutable objects in the same case. I was trying to update de ditcionary `buffer_params` with the new values in `buffer_params_`, but it's better if we use `buffer_params_` in the rest of the code.
Am I right now?

Reply:
It is good now, it is more consistent to use `buffer_params` and `physics_params` in the same footing. I would put the `.copy()` lines one after the other.